pkg/operator/staticpod/controller/prune/prune_controller.go

```diff
@@ -268,10 +283,11 @@ func (c *PruneController) processNextWorkItem() bool {
 }
 
 func (c *PruneController) sync() error {
-	_, operatorStatus, _, err := c.operatorConfigClient.GetStaticPodOperatorState()
+	operatorSpec, operatorStatus, _, err := c.operatorConfigClient.GetStaticPodOperatorState()
 	if err != nil {
 		return err
 	}
+	c.syncRevisionLimits(operatorSpec)
```

Comment:
need an unlimited check

Reply:
unlimited is handled here: https://github.com/openshift/library-go/pull/175/files#diff-5c9aefda4344c1bce9b242c5c5f5d82eR204

Reply:
added to tests too